lib/Charts/ChartType.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+import defined from 'terriajs-cesium/Source/Core/defined';
+
+import LineChart from './LineChart';
+import BarChart from './BarChart';
+
+const chartType = {
+    line: new LineChart(),
+    bar: new BarChart()
+};
+
+function determineChartType(chartData) {
+    
+    // respect chartData.type
+    if (chartType[chartData.type] !== undefined) {
+        return chartType[chartData.type];
+    }
+    
+    // insert hack here for figuring out which type of chart datasets need
+    try
```

Comment:
If you assign a value to the type field in chart data, and it exists, it will use that type of chart. How that integrates with registerCustomComponentTypes.js, will leave future generations to figure out if they need to specify chart types for data.